app/controllers/traces_controller.rb

```diff
@@ -68,10 +68,9 @@ def index
   end
 
   def show
-    @trace = Trace.find(params[:id])
+    @trace = Trace.visible.find(params[:id])
 
-    if @trace&.visible? &&
-       (@trace&.public? || @trace&.user == current_user)
+    if @trace&.public? || @trace&.user == current_user
```

Comment:
I realise it's not a change from this PR but I don't think we need the conditional dereference here do we? The `find` will have raised an exception if nothing was found...

Reply:
replaced `&.` with `.` here